sda-commons-client-jersey/src/test/java/org/sdase/commons/client/jersey/ClientConfiguredProxyTest.java

```diff
@@ -11,59 +11,55 @@
 import com.github.tomakehurst.wiremock.matching.RequestPatternBuilder;
-import io.dropwizard.testing.junit.DropwizardAppRule;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
 import javax.ws.rs.client.Client;
 import javax.ws.rs.core.Response;
 import org.apache.http.HttpHeaders;
 import org.apache.http.HttpStatus;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.rules.RuleChain;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 import org.sdase.commons.client.jersey.test.ClientTestApp;
 import org.sdase.commons.client.jersey.test.ClientTestConfig;
-import org.sdase.commons.server.testing.SystemPropertyRule;
+import org.sdase.commons.client.jersey.wiremock.testing.WireMockClassExtension;
 
 /**
  * A test that checks if the proxy can be configured via the configuration class. They should have a
  * higher priority than the system properties.
  */
-public class ClientConfiguredProxyTest {
-  private static final WireMockClassRule CONTENT_WIRE =
-      new WireMockClassRule(wireMockConfig().dynamicPort());
-
-  private static final WireMockClassRule PROXY_WIRE =
-      new WireMockClassRule(wireMockConfig().dynamicPort());
-
-  // set invalid values to see whether the correct values are used
-  private static final SystemPropertyRule PROP =
-      new SystemPropertyRule()
-          .setProperty("http.proxyHost", "0.0.0.0")
-          .setProperty("http.proxyPort", "12345")
-          .setProperty("http.nonProxyHosts", "localhost");
-
-  private static final DropwizardAppRule<ClientTestConfig> DW =
-      new DropwizardAppRule<>(
+class ClientConfiguredProxyTest {
+  @RegisterExtension
+  @Order(0)
+  private static final WireMockClassExtension CONTENT_WIRE =
+      new WireMockClassExtension(wireMockConfig().dynamicPort());
+
+  @RegisterExtension
+  @Order(1)
+  private static final WireMockClassExtension PROXY_WIRE =
+      new WireMockClassExtension(wireMockConfig().dynamicPort());
+
+  @RegisterExtension
+  private static final DropwizardAppExtension<ClientTestConfig> DW =
+      new DropwizardAppExtension<>(
           ClientTestApp.class,
           resourceFilePath("test-config.yaml"),
           config("client.proxy.host", "localhost"),
           config("client.proxy.port", () -> "" + PROXY_WIRE.port()),
           config("client.proxy.nonProxyHosts", "localhost"));
 
-  @ClassRule
-  public static final RuleChain rule =
-      RuleChain.outerRule(CONTENT_WIRE).around(PROXY_WIRE).around(PROP).around(DW);
-
-  @Before
-  public void before() {
+  @BeforeEach
+  void before() {
     CONTENT_WIRE.resetAll();
     PROXY_WIRE.resetAll();
+    System.setProperty("http.proxyHost", "0.0.0.0");
+    System.setProperty("http.proxyPort", "12345");
+    System.setProperty("http.nonProxyHosts", "localhost");
```

Comment:
Prefer the [SetSystemProperty](https://junit-pioneer.org/docs/system-properties/) extension because it will clean up afterwards.